src/shared/useContract.ts

```diff
@@ -30,19 +41,12 @@ export const useContract = <T extends Abi | Record<string, unknown>>(
     enabled ? key : emptyCacheKey,
     enabled
       ? async () => {
-          const opts =
-            typeof options === 'function'
-              ? options(api)
-              : options || {
-                  gasLimit: api.createType('WeightV2', {
-                    refTime: '100000000000',
-                    proofSize: '1000000',
-                  }),
-                  storageDepositLimit: null,
-                };
+          const opts = getOptions(options, api);
           const response = await contract.query[method](owner, opts, ...(args || []));
           if (!response?.result?.isOk || response?.output === undefined) throw response;
-          return response;
+          // ? TODO: maybe not ideal to cache only output
+          // caching the whole object causes the output to be converted to hex string
```

Comment:
Does this lead to issues?

Reply:
So there is some more investigation I wish to do on this but the problem is that the value received when being cache gets serialized with JSON.stringify fn which converts that value into a hex string. That causes issues when reading value from the cache and trying to convert it back because the result!=cached result.  So this implementation works but would like to put some more time into this to find out if there is a better way.

Reply:
Hmm okay, well I'm fine if we postpone the investigation to another PR if the current solution works for now.

Reply:
Yes the solution works and is satisfactory to the functionality that is needed. But because the hooks is reusable it might be useful to cache the whole result and not just the output part.